lib/frame_repository.rb

```diff
@@ -0,0 +1,21 @@
+#require 'frame'
+
+class FrameRepository
+  def initialize
+      @frame_score= []
+  end
+  
+  def add(roll)
+    while true do
+      #break if roll.index[0] == 10 
+      #return @frame_score << roll.index[0] + 0
+      #else
+      @frame_score << roll
+      break if @frame_score.length <= 2
+   end 
```

Comment:
I don't think this loop is necessary. The first and second time we add a roll, the `@frame_score` array will contain one or two elements and so we break out of the loop immediately. If we try adding a third element to this array, the condition on L14 will evaluate to false, and we will loop infinitely, adding `roll` to `@frame_score` indefinitely.